### PR TITLE
doc: Add basic service documentation

### PR DESCRIPTION
**Pull request by rlawrence25**

### CODE REVIEW

**Review by luke-biel — DISMISSED** [review]

File: docs/Command-Service.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Command Service
+
+### Responsibilities
+
+The command service is reponsible for consuming a specific input topic and writing that data to its to a variety of different stroage solutions.
```
Comment (USSAMOH2) [inline docs/Command-Service.md:5]:
```suggestion
The command service is responsible for consuming a specific input topic and writing that data to a variety of different storage solutions. 
```

File: docs/Command-Service.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Command Service
+
+### Responsibilities
+
+The command service is reponsible for consuming a specific input topic and writing that data to its to a variety of different stroage solutions. 
+
+### Communication
+
+Communication from command service is done through through Kafka. An input topic is consumed and the message is routed to the corresponding db based on configuration.
```
Comment (USSAMOH2) [inline docs/Command-Service.md:9]:
```suggestion
Communication from the command service is done through through Kafka. An input topic is consumed and the message is routed to the corresponding database based on configuration. 
```

File: docs/Data-Router.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+The data router routes requests from RabbitMQ and Kafka to the correct storage solution based on the schema and data type. The data router also communicates with the schema registry based on the `SCHEMA_ID` data is written to a specific topic.
+
+Below is the data needed by data router:
+
+```
+{
+    "schemaId": <UUID>,
+    "objectId": <UUID>,
+    "data": "{ \"some_propery": \"object\"}"
```
Comment (USSAMOH2) [inline docs/Data-Router.md:18]:
```suggestion
    "data": "{ \"some_property": \"object\"}"
```

File: docs/Query-Service.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Query Service
+
+### Responsibilites
+
+The query service is reponsible for querying data from several dabtabases.
```
Comment (USSAMOH2) [inline docs/Query-Service.md:5]:
```suggestion
The query service is responsible for querying data from several databases.
```

File: docs/Query-Service.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+### Communication
+
+Communication to query service is done through [gRPC][grpc] based on two [endpoints][endpoints] of querying for data by `SCHEMA_ID` or multiple `OBJECT_ID`s. Query service communicates with multiple databases such as postgresql, druid, sled. Query service also communicates with [schema registry][schema-registry]. 
+
+
+### Configuration
+
+To configure query service, set environmet variables for `INPUT_PORT`, `DS_QUERY_URL` or `POSTGRES_QUERY_URL` to configure the corressponding database.
```
Comment (USSAMOH2) [inline docs/Query-Service.md:14]:
```suggestion
To configure the query service, set environment variables for `INPUT_PORT`, `DS_QUERY_URL` or `POSTGRES_QUERY_URL` to configure the corresponding database.
```

File: docs/Getting-Started.md [hunk 6]
```diff
@@ -1,15 +1,133 @@
...
+```
+cargo run --bin query_service -- \
+--schema-registry-addr <registry_addr> \
+--ds-query-url <ds-url>  \
+--input-port <input_port>
+```
+
+
+### Query Data via Query Router
+Query Router works with query service to route requests to the correct stoarge solution per schema's query address.
```
Comment (USSAMOH2) [inline docs/Getting-Started.md:110]:
```suggestion
The Query Router works with the query services to route requests to the correct repository, determined per schema (based on its query address).
```

**Review by USSAMOH2 — CHANGES_REQUESTED** [review]

File: docs/Command-Service.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
+# Command Service
+
+### Responsibilities
+
+The command service is responsible for consuming a specific input topic and writing that data to its to a variety of different storage solutions.
```
Comment (Kononnable) [inline docs/Command-Service.md:5]:
```suggestion
The command service is responsible for consuming a specific input topic and writing that data to a variety of different storage solutions. 
```

File: docs/Getting-Started.md [hunk 8]
```diff
@@ -1,15 +1,133 @@
...
+
+- Rust
+- Docker
+- Docker Compose
+
+You can download [docker desktop][docker-desktop] for both Windows and MacOS to intall docker and docker compose on your local machine.
+
+## Working with CDL Locally
+
+Below is a following simple amount of steps to getting started working with the services in the CDL locally on your machine. To build and install container images of services within the cd, run `build.sh` in root directory of this project.
```
Comment (Kononnable) [inline docs/Getting-Started.md:22]:
```suggestion
Below is a following simple amount of steps to getting started working with the services in the CDL locally on your machine. To build and install container images of services within the CDL, run `build.sh` in root directory of this project.
```

File: docs/Getting-Started.md [hunk 9]
```diff
@@ -1,15 +1,133 @@
...
+- Query Data
+
+## Add Schema via CLI
+A schema can be added through the CLI tool localed in the `cdl-cli` directory. To be able to run the cli you must have a rust compiler. The following command below creates the schema with a name according a json schema in a file as well as sets the topic for routing data through kafka. 
+
+```
+cargo run --bin cdl -- --registry-addr <registry_address> schema add --name <schema_name> --topic "cdl.document.input" --file <file_path_to_json>
+```
+
+Here is the typical sample JSON schema format that the CDL anticipates and ultimatley will validate data by. Please review [README][schema-registry] in `schema-registry` directory for more information.
```
Comment (Kononnable) [inline docs/Getting-Started.md:40]:
```suggestion
Here is the sample JSON schema format that the CDL anticipates and ultimatley will validate data by. Please review [README][schema-registry] in `schema-registry` directory for more information.
```

File: docs/Getting-Started.md [hunk 10]
```diff
@@ -1,15 +1,133 @@
...
+    }
+}
+```
+
+**NOTE**: Schema's can be added via [gRPC][grpc] to the schema registry. Ensure that you have `protoc` installed on your machine you machine generate [proto](./schema-registry/proto) files in a supported language and make requests via a client.
+
+
+## Insert Data
+
+Data can be inserted into the system by data being written to Kafka or ingested through RabbitMQ. Data must be in JSON format with the following fields: `schemaId`, `objectId` and `data` to be routed through the CDL.
```
Comment (Kononnable) [inline docs/Getting-Started.md:71]: Right now we listen only on kafka topics, rabbitmq is ready but require few lines of code to be changed.
Question to the team: How we want to approach that? Do we want to enable both by some sort of configuration, or do we keep that as unused feature for now(will be used in the future)?
@epiphany-platform/cdl-team

File: docs/Command-Service.md [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+
+### Configuration
+
+It is important to ensure that configure command service with the url to a database from eitehr Postgres, Druid or SLED
+
+```
+KAFKA_INPUT_GROUP_ID
+KAFKA_INPUT_BROKERS
+KAFKA_INPUT_TOPIC: 
+POSTGRES_OUTPUT_URL
```
Comment (luke-biel) [inline docs/Command-Service.md:20]: Well, this has changed. We had to split POSTGRES config into separate envs for `DE` deployment. 
https://github.com/epiphany-platform/CommonDataLayer/blob/develop/examples/deploy/docker-compose.yml#L103

File: docs/Query-Service.md [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+The query service is responsible for querying data from several databases.
+
+### Communication
+
+Communication to query service is done through [gRPC][grpc] based on two [endpoints][endpoints] of querying for data by `SCHEMA_ID` or multiple `OBJECT_ID`s. Query service communicates with multiple databases such as postgresql, druid, sled. Query service also communicates with [schema registry][schema-registry]. 
+
+
+### Configuration
+
+To configure the query service, set environment variables for `INPUT_PORT`, `DS_QUERY_URL` or `POSTGRES_QUERY_URL` to configure the corresponding database.
```
Comment (luke-biel) [inline docs/Query-Service.md:14]: Same here for postgres
https://github.com/epiphany-platform/CommonDataLayer/blob/develop/examples/deploy/docker-compose.yml#L130

**Review by luke-biel — CHANGES_REQUESTED** [review]

File: docs/Query-Service.md (hunk 12 above)
Comment (esavier) [reply docs/Query-Service.md]: updated, please re-review

File: docs/Command-Service.md [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+- Sleight (CDL's document storage)
+- Troika (CDL's binary data repo)
+- .. or anything with matching GRPC :)
+
+### Configuration (environment files)
+
+```
+KAFKA_INPUT_GROUP_ID
+KAFKA_INPUT_BROKERS
+KAFKA_INPUT_TOPIC:
```
Comment (frondeus) [inline docs/Command-Service.md:30]:
```suggestion
KAFKA_INPUT_TOPIC
```

File: docs/Data-Router.md [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+
+
+### Configuration
+
+To configure data router, set following environment variables. `INPUT_ADDR` and `INPUT_QUEUE` is related to the incoming data in the router. `KAFKA_BROKERS`, `KAFKA_ERROR_CHANNEL` are related to messages being routed through Kafka to the corresponding command service.
+
+
+```
+INPUT_ADDR
+INPUT_QUEUE:
```
Comment (frondeus) [inline docs/Data-Router.md:69]:
```suggestion
INPUT_QUEUE
```

File: docs/Data-Router.md [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+KAFKA_BROKERS:
```
Comment (frondeus) [inline docs/Data-Router.md:70]:
```suggestion
KAFKA_BROKERS
```

File: docs/Data-Router.md [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+KAFKA_ERROR_CHANNEL:
```
Comment (frondeus) [inline docs/Data-Router.md:71]:
```suggestion
KAFKA_ERROR_CHANNEL
```

File: docs/Command-Service.md (hunk 11 above)
Comment (frondeus) [reply docs/Command-Service.md:31]:
```suggestion
POSTGRES_USERNAME
POSTGRES_PASSWORD
POSTGRES_HOST
POSTGRES_PORT
POSTGRES_DBNAME
POSTGRES_SCHEMA
```

File: docs/Query-Service.md [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+### Configuration
+
+`
+INPUT_PORT
+RUST_LOG
+POSTGRES_USERNAME
+POSTGRES_PASSWORD
+POSTGRES_HOST
+POSTGRES_PORT
+POSTGRES_DBNAME
```
Comment (frondeus) [inline docs/Query-Service.md:34]:
```suggestion
POSTGRES_DBNAME
POSTGRES_SCHEMA
```

**Review by frondeus — CHANGES_REQUESTED** [review]

File: docs/Getting-Started.md [hunk 18]
```diff
@@ -1,15 +1,133 @@
...
+-H "Content-Type:application/json" \
+-XPOST -d'{"properties":{},"routing_key":"my_key","payload":"my body","payload_encoding":"string"}'\
+http://${ampq_url}/api/exchanges/%2F/${exchange}/publish
+```
+
+
+## Query Data 
+
+### Query via Query Service
+Following this example local deploymet, you can query for data saved. Here data is saved within the PR. Ensure that environment variables are set for `POSTGRES_QUERY_URL`, `INPUT_PORT` and `DS_QUERY_URL` or run query service directly on machine.
```
Comment (frondeus) [inline docs/Getting-Started.md:99]:
```suggestion
Following this example local deploymet, you can query for data saved. Here data is saved within the PR. Ensure that environment variables are set for `POSTGRES_USERNAME`, `POSTGRES_PASSWORD`, `POSTGRES_HOST`, `POSTGRES_PORT`, `POSTGRES_DBNAME`, `POSTGRES_SCHEMA`, `INPUT_PORT` and `DS_QUERY_URL` or run query service directly on machine.
```

**Review by luke-biel — APPROVED** [review]

**Pull request merged**